### PR TITLE
[mob][packages] Add popup menu component

### PR DESCRIPTION
**Pull request by AmanRajSinghMourya**
## Summary
- Add a shared popup menu component to ente_components.
- Add popup menu variations to the component catalog example.
- Add widget tests for sizing, selection, empty options, and async disposal.

### CODE REVIEW

File: mobile/packages/ente_components/lib/components/popup_menu_component.dart [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+  final Widget? secondaryTrailingWidget;
+  final Color? labelColor;
+  final Widget? leadingWidget;
+  final bool isActive;
+  final Widget? trailingWidget;
+  final Widget? activeTrailingWidget;
+  final bool showDivider;
+}
+
+class EntePopupMenuButton<T> extends StatelessWidget {
```
[the remark this answers is not in the document]
Comment (AmanRajSinghMourya) [reply mobile/packages/ente_components/lib/components/popup_menu_component.dart:42]: Yes there will be a follow up PR to solve this

[the remark this answers is not in the document]
Comment (AmanRajSinghMourya) [reply mobile/packages/ente_components/lib/components/popup_menu_component.dart:42]: #11071 Have added here

**Review by ashilkn — APPROVED** [review]
